lib/IRGen/IRGenSIL.cpp

```diff
@@ -3896,6 +3896,11 @@ static bool hasReferenceSemantics(IRGenSILFunction &IGF,
 static llvm::Value *emitIsUnique(IRGenSILFunction &IGF, SILValue operand,
                                  SourceLoc loc, bool checkPinned) {
   if (!hasReferenceSemantics(IGF, operand->getType())) {
+    if (IGF.IGM.DebugInfo &&
+        IGF.IGM.IRGen.Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView)
+      // In CodeView, ``llvm.trap`` should be at an artificial location.
+      IGF.IGM.DebugInfo->setArtificialTrapLocation(IGF.Builder,
+                                                   IGF.getDebugScope());
     IGF.emitTrap(/*EmitUnreachable=*/false);
```

Comment:
Have you considered sinking the setArtificialTrapLocation call into emitTrap, so that new uses of emitTrap are correct by default?